test/gtest/ucp/test_ucp_ep_reconfig.cc

```diff
@@ -103,60 +113,60 @@ class test_ucp_ep_reconfig : public ucp_test {
+                      const mem_buffer *rbuf, size_t msg_size,
                       std::vector<void*> &reqs)
     {
         const ucp_request_param_t param = {
             .op_attr_mask = UCP_OP_ATTR_FLAG_NO_IMM_CMPL
         };
 
-        void *sreq = ucp_tag_send_nbx(e1.ep(), sbuf.ptr(), msg_size, 0, &param);
-        void *sreq_sync = ucp_tag_send_sync_nbx(e1.ep(), sbuf.ptr(), msg_size,
+        void *sreq = ucp_tag_send_nbx(e1.ep(), sbuf->ptr(), msg_size, 0,
+                                      &param);
+        void *sreq_sync = ucp_tag_send_sync_nbx(e1.ep(), sbuf->ptr(), msg_size,
                                                 0, &param);
         reqs.insert(reqs.end(), {sreq, sreq_sync});
 
         for (unsigned iter = 0; iter < 2; iter++) {
-            void *rreq = ucp_tag_recv_nbx(e2.worker(), rbuf.ptr(), msg_size, 0,
-                                          0, &param);
+            void *rreq = ucp_tag_recv_nbx(e2.worker(), rbuf->ptr(), msg_size,
+                                          0, 0, &param);
             reqs.push_back(rreq);
         }
     }
 
     void send_recv(bool bidirectional)
     {
-        static const unsigned num_iterations = 1000;
 /* TODO: remove this when 100MB asan bug is solved */
 #ifdef __SANITIZE_ADDRESS__
         static const size_t msg_sizes[] = {8, 1024, 16384, 65536};
 #else
         static const size_t msg_sizes[] = {8, 1024, 16384, UCS_MBYTE};
 #endif
+        mem_buffer_vec_t sbufs, rbufs, o_sbufs, o_rbufs;
 
-        for (auto msg_size : msg_sizes) {
-            for (unsigned i = 0; i < num_iterations; ++i) {
-                mem_buffer sbuf(msg_size, UCS_MEMORY_TYPE_HOST, i);
-                mem_buffer rbuf(msg_size, UCS_MEMORY_TYPE_HOST, ucs::rand());
-                mem_buffer o_sbuf(msg_size, UCS_MEMORY_TYPE_HOST, i);
-                mem_buffer o_rbuf(msg_size, UCS_MEMORY_TYPE_HOST, ucs::rand());
-
-                std::vector<void*> reqs;
-                send_message(sender(), receiver(), sbuf, rbuf, msg_size, reqs);
+        init_buffers(sbufs, rbufs, o_sbufs, o_rbufs,
+                     msg_sizes[ucs_static_array_size(msg_sizes)-1],
+                     bidirectional);
 
-                if (bidirectional) {
-                    send_message(receiver(), sender(), o_sbuf, o_rbuf, msg_size,
-                                 reqs);
-                }
+        for (auto msg_size : msg_sizes) {
+            std::vector<void*> reqs;
 
-                requests_wait(reqs);
-                rbuf.pattern_check(i);
+            for (unsigned i = 0; i < num_iterations; ++i) {
+                send_message(sender(), receiver(), sbufs[i].get(),
+                             rbufs[i].get(), msg_size, reqs);
 
                 if (bidirectional) {
-                    o_rbuf.pattern_check(i);
+                    send_message(receiver(), sender(), o_sbufs[i].get(),
+                                 o_rbufs[i].get(), msg_size, reqs);

```

Comment:
maybe instead of using separate vectors for the other-way buffers, we can allocate 2x the number of buffers for bidirectional tests, and here use something like 
```c
send_message(receiver(), sender(), sbufs[i + num_iterations].get(),
             o_rbufs[i + num_iterations].get(), msg_size, reqs);
```